packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -137,13 +164,78 @@ export const useGridKeyboardNavigation = (
     ],
   );
 
+  const navigateColumnHeders = React.useCallback(
+    (params: GridCellParams, event: React.KeyboardEvent) => {
+      event.preventDefault();
+      let nextColumnHeaderIndexes: GridColumnHeaderIndexCoordinates | null;
+      const { colIndex } = params;
+      const key = mapKey(event);
+      const ColumnHeaderEl = params.element!;
+      ColumnHeaderEl.tabIndex = -1;
+
+      if (isArrowKeys(key)) {
+        nextColumnHeaderIndexes = getNextColumnHeaderIndexes(key, {
+          colIndex,
+        });
+      } else if (isHomeOrEndKeys(key)) {
+        const colIdx = key === 'Home' ? 0 : colCount - 1;
+
+        nextColumnHeaderIndexes = { colIndex: colIdx };
+      } else if (isPageKeys(key)) {
+        // Handle only Page Down key, Page Up should keep the current possition
+        if (key.indexOf('Down') > -1) {
+          apiRef.current.setCellFocus({
+            colIndex: params.colIndex,
+            rowIndex: containerSizes!.viewportPageSize - 1,
+          });
+        }
+        return;
+      } else {
+        throw new Error('Material-UI. Key not mapped to navigation behavior.');
+      }
+
+      if (!nextColumnHeaderIndexes) {
+        apiRef.current.setCellFocus({ colIndex: params.colIndex, rowIndex: 0 });
+        return;
+      }
+
+      nextColumnHeaderIndexes!.colIndex =
+        nextColumnHeaderIndexes!.colIndex <= 0 ? 0 : nextColumnHeaderIndexes!.colIndex;
```

Comment:
Seems simpler to read and to identify patterns later on (clamp):

```suggestion
      nextColumnHeaderIndexes!.colIndex = Math.min(0, nextColumnHeaderIndexes!.colIndex);
```

x Math.max

Reply:
https://github.com/mui-org/material-ui/blob/bcb61849635b44c44c687a11e12a96087d582397/packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js#L25

Reply:
To have it working like this it will look more like that:
`nextColumnHeaderIndexes!.colIndex = Math.max(0, nextColumnHeaderIndexes!.colIndex);`